pkg/disttask/framework/taskexecutor/execute/interface.go

```diff
@@ -63,9 +63,65 @@ type StepExecutor interface {
 	ResourceModified(ctx context.Context, newResource *proto.StepResource) error
 }
 
-// SubtaskSummary contains the summary of a subtask.
+// SubtaskSummary contains the summary of a subtask
+// These fields represent the number of data/rows inputed to the subtask.
 type SubtaskSummary struct {
-	RowCount int64
+	RowCnt int64 `json:"row_count"`
+	Bytes  int64 `json:"bytes"`
+}
+
+// RunningSubtaskSummary is used to store the summary of a running subtask.
+type RunningSubtaskSummary struct {
+	RowCnt atomic.Int64
+	Bytes  atomic.Int64
+}
+
+// ResetMetrics resets the summary to the given row count and bytes.
+func (s *RunningSubtaskSummary) ResetMetrics() {
+	s.RowCnt.Store(0)
+	s.Bytes.Store(0)
+}
+
+// ToSummary converts the running subtask summary to a subtask summary.
+func (s *RunningSubtaskSummary) ToSummary() *SubtaskSummary {
+	return &SubtaskSummary{
+		RowCnt: s.RowCnt.Load(),
+		Bytes:  s.Bytes.Load(),
+	}
+}
+
+// Collector is the interface for collecting subtask metrics.
+type Collector interface {
+	Add(bytes, rows int64)
+}
+
+// collector is the implement of Collector interface
+type collector struct {
+	addFunc func(bytes, rows int64)
+}
```

Comment:
maybe have different impl of `Collector` for import and add-index, no need to use function as field